src/ext.ml

```diff
@@ -60,31 +60,76 @@ module Context = struct
     | _ -> assert ((T a) <> (T b)); Ne
 
   let get_extension : type a. a t -> a -> (extension * attributes) option = fun t x ->
-    match t, x with
-    | Class_expr       , {pcl_desc =Pcl_extension  e; pcl_attributes =a;_} -> Some (e, a)
-    | Class_field      , {pcf_desc =Pcf_extension  e; pcf_attributes =a;_} -> Some (e, a)
-    | Class_type       , {pcty_desc=Pcty_extension e; pcty_attributes=a;_} -> Some (e, a)
-    | Class_type_field , {pctf_desc=Pctf_extension e; pctf_attributes=a;_} -> Some (e, a)
-    | Core_type        , {ptyp_desc=Ptyp_extension e; ptyp_attributes=a;_} -> Some (e, a)
-    | Expression       , {pexp_desc=Pexp_extension e; pexp_attributes=a;_} -> Some (e, a)
-    | Module_expr      , {pmod_desc=Pmod_extension e; pmod_attributes=a;_} -> Some (e, a)
-    | Module_type      , {pmty_desc=Pmty_extension e; pmty_attributes=a;_} -> Some (e, a)
-    | Pattern          , {ppat_desc=Ppat_extension e; ppat_attributes=a;_} -> Some (e, a)
-    | Signature_item   , {psig_desc=Psig_extension(e, a)               ;_} -> Some (e, a)
-    | Structure_item   , {pstr_desc=Pstr_extension(e, a)               ;_} -> Some (e, a)
-    | _ -> None
+    match t with
+    | Class_expr       ->
```

Comment:
Would it be worth extracting view functions here to simplify this into a single `match%view` ?

Reply:
If we had `[@@deriving view]` I'd do it. Writing them manually seems not worth it. But I think we need `[@@deriving view]` eventually, so that clients can write things like this easily. For now I'm inclined to just use nested matches here.